enigma-principal/app/src/epoch_u/epoch_provider.rs

```diff
@@ -30,64 +30,60 @@ use std::mem::replace;
+        Ok(mut f) => {
+            let mut buf = Vec::new();
+            f.read_to_end(&mut buf)?;
+            let mut des = Deserializer::new(&buf[..]);
+            let mut data: Vec<EpochState> = Deserialize::deserialize(&mut des).unwrap_or_default();
+            println!("Found EpochState list: {:?}", data);
+            if cap < data.len() {
+                return Err(EpochStateIOErr { message: format!("The EpochState entries exceed the cap: {}", cap) }.into());
             }
-        };
-        path.push("epoch-state.msgpack");
-        Ok(path)
-    }
+            let mut capped_data = Vec::with_capacity(cap);
+            capped_data.append(&mut data);
+            capped_data
+        }
+        Err(_) => {
+            println!("No existing epoch state, starting with block 0");
+            vec![]
+        }
+    };
+    Ok(epoch_state)
+}
 
-    #[logfn(DEBUG)]
-    fn read_from_file(cap: usize) -> Result<Vec<EpochState>, Error> {
-        let epoch_state = match File::open(Self::get_file_path()?) {
-            Ok(mut f) => {
-                let mut buf = Vec::new();
-                f.read_to_end(&mut buf)?;
-                let mut des = Deserializer::new(&buf[..]);
-                let mut data: Vec<EpochState> = Deserialize::deserialize(&mut des).unwrap_or_default();
-                println!("Found EpochState list: {:?}", data);
-                if cap < data.len() {
-                    return Err(EpochStateIOErr { message: format!("The EpochState entries exceed the cap: {}", cap) }.into());
-                }
-                let mut capped_data = Vec::with_capacity(cap);
-                capped_data.append(&mut data);
-                capped_data
-            }
-            Err(_) => {
-                println!("No existing epoch state, starting with block 0");
-                vec![]
-            }
-        };
-        Ok(epoch_state)
+#[logfn(DEBUG)]
+fn write_to_file(path: PathBuf, epoch_state: Vec<EpochState>) -> Result<(), Error> {
+    let path = get_file_path(path)?;
+    if epoch_state.is_empty() {
+        return Ok(fs::remove_file(path).unwrap_or_else(|_| println!("No epoch state file to remove")));
     }
+    let mut file = File::create(path)?;
+    let mut buf = Vec::new();
+    epoch_state.serialize(&mut Serializer::new(&mut buf))?;
+    file.write_all(&buf)?;
+    Ok(())
+}
 
-    #[logfn(DEBUG)]
-    fn write_to_file(epoch_state: Vec<EpochState>) -> Result<(), Error> {
-        let path = Self::get_file_path()?;
-        if epoch_state.is_empty() {
-            return Ok(fs::remove_file(path).unwrap_or_else(|_| println!("No epoch state file to remove")));
-        }
-        let mut file = File::create(path)?;
-        let mut buf = Vec::new();
-        epoch_state.serialize(&mut Serializer::new(&mut buf))?;
-        file.write_all(&buf)?;
-        Ok(())
+impl EpochStateManager {
+    pub fn new(mut dir_path: PathBuf, cap: usize) -> Result<Self, Error> {
```

Comment:
Why EpochStateManager holds directory and not file. Why it is always needed to pass through `get_file_path`?